buildkit/extraction.py

```diff
@@ -90,6 +92,8 @@ def prune_dir(unpack_root, ignore_files):
 
 def _extract_tar_with_7z(binary, archive_path, output_dir, relative_to):
     get_logger().debug('Using 7-zip extractor')
+    if os.path.exists(output_dir / relative_to):
+        shutil.rmtree(output_dir / relative_to)
```

Comment:
I get an error if the extraction failed. So I think it's better to clean this folder before extraction ?

Reply:
It is the user's responsibility to clean up the source tree after a failed attempt. Removing files preemptively is dangerous if the user makes a mistake with their paths. At most, I would tell the user what happen and recommend a course of action.

Also, the error that you get is probably the one right below this code you added...

(Also FYI, this code breaks under Python 3.5 because it doesn't implement PEP 519)